### PR TITLE
RDM-7282 Introduce BEFTA FW

### PR DESCRIPTION
**Pull request by RebeccaBaker**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-7282

### Change description ###
Introduce befta framework in prep for functional tests

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/aat/resources/features/F-000/S-000-Base-Root.td.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+		}
+	},
+	"request": {
+		"headers": {
+			"Authorization": "[[DYNAMIC]]",
+			"ServiceAuthorization": "[[DYNAMIC]]",
+			"Content-Type": "application/json"
+		}
+	}
+}
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-000/S-000-Base-Root.td.json]: This one doesn't look necessary.

File: src/aat/resources/features/F-000/S-000-Base-Success.td.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+   "_guid_": "S-000-Base-Success",
+   "_extends_": "S-000-Base-Root",
+
+   "expectedResponse": {
+      "responseCode": 200,
+      "responseMessage": "OK"
+   }
+}
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-000/S-000-Base-Success.td.json]: This one doesn't look necessary.

File: Jenkinsfile_CNP [hunk 3]
```diff
@@ -11,8 +11,28 @@ def product = "ccd"
...
+        secret('ccd-private-caseworker-solicitor-email', 'CCD_PRIVATE_CASEWORKER_SOLICITOR_EMAIL'),
+        secret('ccd-private-caseworker-solicitor-password', 'CCD_PRIVATE_CASEWORKER_SOLICITOR_PASSWORD'),
+        secret('ccd-private-cross-case-type-worker-email', 'CCD_PRIVATE_CROSS_CASE_TYPE_CASEWORKER_EMAIL'),
+        secret('ccd-private-cross-case-type-caseworker-password', 'CCD_PRIVATE_CROSS_CASE_TYPE_CASEWORKER_PASSWORD'),
+        secret('ccd-private-cross-case-type-solicitor-email', 'CCD_PRIVATE_CROSS_CASE_TYPE_SOLICITOR_EMAIL'),
+        secret('ccd-private-cross-case-type-solicitor-password', 'CCD_PRIVATE_CROSS_CASE_TYPE_SOLICITOR_PASSWORD'),
+        secret('ccd-restricted-cross-case-type-caseworker-email', 'CCD_RESTRICTED_CROSS_CASE_TYPE_CASEWORKER_EMAIL'),
+        secret('ccd-restricted-cross-case-type-caseworker-password', 'CCD_RESTRICTED_CROSS_CASE_TYPE_CASEWORKER_PASSWORD'),
+        secret('ccd-api-gateway-oauth2-client-secret', 'OAUTH2_CLIENT_SECRET'),
+    ],
```
Comment (MSancaktutar) [inline Jenkinsfile_CNP]: Most of these will not be needed in the FTA of this particular application, as it's API is relatively much simpler without requiring this many different users. Could we just keep only the ones we are sure will be needed? If later automations require more of these, they can be added later as and when needed.

File: build.gradle [hunk 4]
```diff
@@ -175,6 +177,10 @@ dependencies {
     testCompile group: 'com.opentable.components', name: 'otj-pg-embedded', version: '0.13.1'
     testCompile group: 'com.github.tomakehurst', name: 'wiremock-standalone', version: '2.21.0'
     testCompile group: 'org.springframework.cloud', name: 'spring-cloud-contract-wiremock', version: '2.1.0.RELEASE'
+
+    compile group: 'io.cucumber', name: 'cucumber-java', version: '5.0.0-RC1'
+    testCompile group: 'io.cucumber', name: 'cucumber-junit', version: '5.0.0-RC1'
+    testCompile files('lib/befta-fw-1.0.9-SNAPSHOT.jar')
```
Comment (MSancaktutar) [inline build.gradle]: Could we please use the latest available version?

File: Jenkinsfile_CNP (hunk 3 above)
Comment (RebeccaBaker) [reply Jenkinsfile_CNP]: done

File: Jenkinsfile_CNP [hunk 5]
```diff
@@ -11,8 +11,14 @@ def product = "ccd"
...
 def secrets = [
+    'ccd-${env}': [
+        secret('ccd-caseworker-autotest-email', 'CCD_CASEWORKER_AUTOTEST_EMAIL'),
+        secret('ccd-caseworker-autotest-password', 'CCD_CASEWORKER_AUTOTEST_PASSWORD'),
+        secret('ccd-api-gateway-oauth2-client-secret', 'OAUTH2_CLIENT_SECRET'),
+    ],
     's2s-${env}': [
-        secret('microservicekey-ccd-data', 'CCD_DS_SERVICE_SECRET')
+        secret('microservicekey-ccd-data', 'CCD_DS_SERVICE_SECRET'),
+        secret('microservicekey-ccd-gw', 'CCD_GW_SERVICE_SECRET')
```
Comment (MSancaktutar) [inline Jenkinsfile_CNP]: These 2 can be replaced with only: BEFTA_S2S_CLIENT_SECRET

File: Jenkinsfile_CNP [hunk 6]
```diff
@@ -30,8 +36,13 @@ def vaultOverrides = [
     'spreview': 'saat'
 ]
 // Vars needed for AKS functional and smoke tests
+env.IDAM_URL = "https://idam-api.aat.platform.hmcts.net"
 env.S2S_URL = "http://rpe-service-auth-provider-aat.service.core-compute-aat.internal"
 env.CCD_DS_SERVICE_NAME = "ccd_data"
+env.CCD_GW_SERVICE_NAME = "ccd_gw"
```
Comment (MSancaktutar) [inline Jenkinsfile_CNP]: These 2 can be replaced by only: BEFTA_S2S_CLIENT_ID.
The value will be 'ccd_data'.

**Comment by noronhaa** [comment]
Not sure how relevant this is but I see version 1.0.9 jar file in the lib dir for the automation framework dependency does a later version need to be built/used in this PR ?

**Pull request merged**